File: legacy/service-department/src/domain/partner/models/partner.py
""" This module has the definition of the Partner entity
"""

from dataclasses import asdict, dataclass
from datetime import datetime
from enum import Enum
from uuid import UUID, uuid4
from src.domain.partner.models.branch import Branch
from src.domain.partner.models.bank_account import BankAccount
from src.domain.partner.models.user_profile import UserProfile

from src.utils.dict_factory import asdict_factory


class PartnerCategory(Enum):
    ELECTRONICS = "ELECTRONICS"
    FASHION = "FASHION"
    FURNITURE = "FURNITURE"
    CERAMICS_AND_SANITARY_WARE = "CERAMICS_AND_SANITARY_WARE"
    AUTO_SPARE_PARTS = "AUTO_SPARE_PARTS"
    BABY_AND_TOYS = "BABY_AND_TOYS"
    JEWELRY = "JEWELRY"
    SUPERMARKETS = "SUPERMARKETS"
    EVENT_PLANNING = "EVENT_PLANNING"
    EDUCATION = "EDUCATION"
    MOTORCYCLES = "MOTORCYCLES"
    HOME_WARE = "HOME_WARE"
    TOURISM_AND_ENTERTAINMENT = "TOURISM_AND_ENTERTAINMENT"
    MEDICAL = "MEDICAL"
    ACCESSORIES = "ACCESSORIES"
    WEDDING_HALLS = "WEDDING_HALLS"
    OPTICS = "OPTICS"
    SPORTS = "SPORTS"
    FINISHING = "FINISHING"
    MOBILE = "MOBILE"
    SHOPPING_HUBS = "SHOPPING_HUBS"


class PartnerStatus(Enum):
    ACTIVE = "ACTIVE"
    DISABLED = "DISABLED"


@dataclass
class Partner:
    """Definition of the Partner entity"""

    id: UUID
    name: str
    categories: list[PartnerCategory]
    status: PartnerStatus
    tax_registration_number: str
    commercial_registration_number: str
    created_at: datetime
    updated_at: datetime
    branches: list[Branch]
    bank_accounts: list[BankAccount]
    user_profiles: list[UserProfile]

    def __init__(
        self,
        name: str,
        categories: list[PartnerCategory],
        tax_registration_number: str,
        commercial_registration_number: str,
        branches: list[Branch] = [],
        bank_accounts: list[BankAccount] = None,
        user_profiles: list[UserProfile] = None,
        status: PartnerStatus = PartnerStatus.ACTIVE,
        id: UUID = None,
    ):
        self.id = uuid4() if id is None else id
        self.name = name
        self.categories = categories
        self.status = status
        self.tax_registration_number = tax_registration_number
        self.commercial_registration_number = commercial_registration_number
        self.created_at = datetime.now()
        self.updated_at = datetime.now()
        self.branches = branches
        self.bank_accounts = bank_accounts if bank_accounts is not None else []
        self.user_profiles = user_profiles if user_profiles is not None else []

    def to_dict(self):
        result = asdict(self, dict_factory=asdict_factory)
        return result

File: legacy/bff/src/domain/partner/models/partner.ts
export const enCategory: { [key: string]: string } = {
  ALL: "All",
  ELECTRONICS: "Home Appliances",
  FASHION: "Fashion",
  HOME_WARE: "Home ware",
  FURNITURE: "Furniture",
  FINISHING: "Finishing",
  CERAMICS_AND_SANITARY_WARE: "Ceramics & sanitary ware",
  AUTO_SPARE_PARTS: "Auto spare parts",
  BABY_AND_TOYS: "Baby & toys",
  JEWELRY: "Jewelry",
  SUPERMARKETS: "Supermarkets",
  EVENT_PLANNING: "Event planning",
  EDUCATION: "Education",
  MOTORCYCLES: "Motorcycles",
  TOURISM_AND_ENTERTAINMENT: "Tourism & Entertainment",
  MEDICAL: "Medical",
  ACCESSORIES: "Accessories",
  WEDDING_HALLS: "Wedding Halls",
  OPTICS: "Optics",
  SPORTS: "Sports",
  MOBILE: "Mobile & Electronics",
  SHOPPING_HUBS: "Shopping Hubs"
};

export const arCategory: { [key: string]: string } = {
  ALL: "الكل",
  ELECTRONICS: "أجهزة منزلية",
  HOME_WARE: "أدوات منزلية",
  FASHION: "أزياء وملابس",
  FURNITURE: "أثاث وديكور",
  FINISHING: "تشطيبات",
  CERAMICS_AND_SANITARY_WARE: "سيراميك وأدوات صحية",
  AUTO_SPARE_PARTS: "قطع غيار سيارات",
  BABY_AND_TOYS: "أطفال وألعاب",
  JEWELRY: "مجوهرات",
  SUPERMARKETS: "بقالة وسوبر ماركت",
  EVENT_PLANNING: "أفراح ومناسبات",
  EDUCATION: "تعليم",
  MOTORCYCLES: "موتوسيكلات",
  TOURISM_AND_ENTERTAINMENT: "سياحة و ترفيه",
  MEDICAL: "خدمات طبية",
  ACCESSORIES: "إكسسوارات",
  WEDDING_HALLS: "قاعات أفراح",
  OPTICS: "بصريات",
  SPORTS: "رياضة",
  MOBILE: "موبايلات والكترونيات",
  SHOPPING_HUBS: "مراكز تسوق"
};

export const categoriesIcons: { [key: string]: string } = {
  ALL: "",
  ELECTRONICS: "electronics.svg",
  FASHION: "fashion.svg",
  FURNITURE: "furniture.svg",
  FINISHING: "finishing.svg",
  HOME_WARE: "homeware.svg",
  CERAMICS_AND_SANITARY_WARE: "ceramics.svg",
  AUTO_SPARE_PARTS: "automotive.svg",
  BABY_AND_TOYS: "kids.svg",
  JEWELRY: "jewelry.svg",
  SUPERMARKETS: "grocery.svg",
  EVENT_PLANNING: "events.svg",
  EDUCATION: "school.svg",
  MOTORCYCLES: "motorcycle.svg",
  TOURISM_AND_ENTERTAINMENT: "tourism.svg",
  MEDICAL: "medical.svg",
  ACCESSORIES: "accessories.svg",
  WEDDING_HALLS: "weddinghalls.svg",
  OPTICS: "optics.svg",
  SPORTS: "sports.svg",
  MOBILE: "mobile.svg",
  SHOPPING_HUBS: "hubs.svg"
};

export type category = {
  name: string;
  slug?:string;
  icon: string;
};
